### ISSUE TITLE
how to use checkpoint to continue training?

### ISSUE THREAD

**Issue by jiachen0212** [vita-epfl/openpifpaf#81]
hell0 @svenkreiss , first of all, thanks to your work.
emm... I wonder how to use the checkpoint to continue training...
i use:
......
--lr-decay 60 70 \
  --batch-size=8 \
  --basenet=resnet50block5 \
  --head-quad=1 \
  --headnets pif paf \
  --square-edge=401 \
  --regression-loss=laplace \
  --lambdas 30 2 2 50 3 3 \
  --freeze-base=1
  --checkpoint='outputs/resnet50block5-pif-paf-edge401-190613-085512.pkl.epoch008'

but it does not work..    
hope you can give some help...

**Comment by jiachen0212** [comment]
> hell0 @svenkreiss , first of all, thanks to your work.
> emm... I wonder how to use the checkpoint to continue training...
> i use:
> ......
> --lr-decay 60 70 
> --batch-size=8 
> --basenet=resnet50block5 
> --head-quad=1 
> --headnets pif paf 
> --square-edge=401 
> --regression-loss=laplace 
> --lambdas 30 2 2 50 3 3 
> --freeze-base=1
> --checkpoint='outputs/resnet50block5-pif-paf-edge401-190613-085512.pkl.epoch008'
> 
> but it does not work..
> hope you can give some help...

i got it.....   do some change in net.py~~~

**Comment by svenkreiss** [comment]
Thanks for reporting this.

You had to change `net.py` to get it to work? It should have worked without changing code. 

Can you provide some details to see whether this is a bug that needs to be fixed?

**Comment by jiachen0212** [comment]
@svenkreiss  wow, thanks for reply~~~
i did the change like this:
1. put the right checkpoint path in net.py 
2. in train.py, i change some code at around line 150(under the def main():):
org:
        for i in range(-args.freeze_base, 0):
            ftrainer.train(pre_train_loader, i)
me:
        for i in range(-args.freeze_base, 0):
            ftrainer.train(pre_train_loader, i + (start_epoch + 1))

then, it looks like worked...

I am not sure if this is correct, hope you can give pointers. Thank you~

**Comment by svenkreiss** [comment]
When training from a checkpoint, remove the `--freeze-base` argument from your command. It should work without code changes.

**Comment by jiachen0212** [comment]
> When training from a checkpoint, remove the `--freeze-base` argument from your command. It should work without code changes.

ok  thanks  i will try it ~~~

**Comment by jiachen0212** [comment]
@svenkreiss    emmm,,,   i tried the method, but the lr looks a bit wrong:

INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 30, 'n_batches': 9704, 'time': 0.265, 'data_time': 0.0, 'lr': 0.001, 'loss': 0.609, 'head_losses': [0.002, 0.046, 0.005, 0.001, 0.055, 0.072]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 40, 'n_batches': 9704, 'time': 0.264, 'data_time': 0.0, 'lr': 0.001, 'loss': 0.615, 'head_losses': [0.002, 0.056, 0.006, 0.001, 0.065, 0.063]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 50, 'n_batches': 9704, 'time': 0.263, 'data_time': 0.278, 'lr': 0.001, 'loss': 2.09, 'head_losses': [0.008, 0.204, 0.012, 0.007, 0.162, 0.201]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 60, 'n_batches': 9704, 'time': 0.264, 'data_time': 0.278, 'lr': 0.001, 'loss': 0.095, 'head_losses': [0.001, 0.009, 0.001, 0.0, 0.006, 0.008]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 70, 'n_batches': 9704, 'time': 0.266, 'data_time': 0.313, 'lr': 0.001, 'loss': 0.832, 'head_losses': [0.002, 0.055, 0.009, 0.002, 0.084, 0.097]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 80, 'n_batches': 9704, 'time': 0.266, 'data_time': 0.585, 'lr': 0.001, 'loss': 0.213, 'head_losses': [0.001, 0.027, 0.002, 0.001, 0.015, 0.019]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 90, 'n_batches': 9704, 'time': 0.272, 'data_time': 0.694, 'lr': 0.001, 'loss': 0.825, 'head_losses': [0.003, 0.068, 0.006, 0.003, 0.07, 0.078]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 100, 'n_batches': 9704, 'time': 0.257, 'data_time': 0.203, 'lr': 0.001, 'loss': 0.015, 'head_losses': [0.0, None, 0.0, 0.0, None, None]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 110, 'n_batches': 9704, 'time': 0.266, 'data_time': 1.185, 'lr': 0.001, 'loss': 4.309, 'head_losses': [0.008, 0.249, 0.033, 0.012, 0.466, 0.498]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 120, 'n_batches': 9704, 'time': 0.265, 'data_time': 1.261, 'lr': 0.001, 'loss': 1.59, 'head_losses': [0.003, 0.11, 0.014, 0.004, 0.161, 0.191]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 130, 'n_batches': 9704, 'time': 0.265, 'data_time': 1.091, 'lr': 0.001, 'loss': 0.239, 'head_losses': [0.001, 0.03, 0.001, 0.001, 0.02, 0.02]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 140, 'n_batches': 9704, 'time': 0.267, 'data_time': 1.15, 'lr': 0.001, 'loss': 2.882, 'head_losses': [0.007, 0.216, 0.027, 0.007, 0.305, 0.314]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 150, 'n_batches': 9704, 'time': 0.267, 'data_time': 1.176, 'lr': 0.001, 'loss': 0.213, 'head_losses': [0.001, 0.024, 0.001, 0.001, 0.016, 0.017]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 160, 'n_batches': 9704, 'time': 0.268, 'data_time': 1.027, 'lr': 0.001, 'loss': 0.108, 'head_losses': [0.001, 0.013, 0.001, 0.0, 0.007, 0.01]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 170, 'n_batches': 9704, 'time': 0.283, 'data_time': 1.102, 'lr': 0.001, 'loss': 0.56, 'head_losses': [0.002, 0.052, 0.005, 0.001, 0.042, 0.067]}
INFO:Trainer:{'type': 'train', 'epoch': 42, 'batch': 180, 'n_batches': 9704, 'time': 0.269, 'data_time': 1.056, 'lr': 0.001, 'loss': 0.04, 'head_losses': [0.0, 0.003, 0.0, 0.0, 0.001, 0.001]}

lr==0.001???  maybe ,,,,  still need some change in the code...

**Comment by svenkreiss** [comment]
0.001 is the default learning rate after warm-up. And because you are running from a checkpoint, the warm-up already happened.

**Comment by jiachen0212** [comment]
@svenkreiss   ok thanks for your reply ~~~ i real need some change about the lr ~
